### PR TITLE
Allow changing MAX_FRAME_SIZE.

### PR DESCRIPTION
**Pull request by Lukasa**
Motivation:

SETTINGS_MAX_FRAME_SIZE is changeable: we should allow that change to
occur.

Modifications:

- Added fields for communicating MAX_FRAME_SIZE changes.
- Update the encoders/decoders as those settings change.
- Throw errors from the encoders/decoders if violated.

Result:

Users and remote peers can change MAX_FRAMES_SIZE.

### CODE REVIEW

File: Sources/NIOHTTP2/HTTP2FrameParser.swift [hunk 1]
```diff
@@ -923,6 +929,7 @@ struct HTTP2FrameDecoder {
 struct HTTP2FrameEncoder {
     private let allocator: ByteBufferAllocator
     var headerEncoder: HPACKEncoder
+    var maxFrameSize: UInt32 = 1<<14
```
Comment (normanmaurer) [inline Sources/NIOHTTP2/HTTP2FrameParser.swift]: I am not against using `<<` but I think most of the times people prefer to have the "real value" here. If you want to use `<<` maybe add some link / comment why this is a good default value.

File: Sources/NIOHTTP2/HTTP2FrameParser.swift [hunk 2]
```diff
@@ -227,6 +227,7 @@ struct HTTP2FrameDecoder {
     internal var headerDecoder: HPACKDecoder
     private var state: ParserState
     private var allocator: ByteBufferAllocator
+    internal var maxFrameSize: UInt32 = 1<<14
```
Comment (normanmaurer) [inline Sources/NIOHTTP2/HTTP2FrameParser.swift]: same as below

File: Sources/NIOHTTP2/HTTP2FrameParser.swift (hunk 1 above)
Comment (Lukasa) [reply Sources/NIOHTTP2/HTTP2FrameParser.swift]: Heh, it's a good default value because it's what the spec requires. ;) Will add.

Comment (normanmaurer) [reply Sources/NIOHTTP2/HTTP2FrameParser.swift]: Thats what I assumed :D So adding a link to the spec is what we need :)

**Comment by Lukasa** [comment]
Updated.

**Pull request merged**